### PR TITLE
Added API arguments to main file functions plus console script

### PR DESCRIPTION
**Pull request by ArneBachmann**
Since there was no testing code in the repository, I cannot check correctness.
Please test and merge as you see fit.

### CODE REVIEW

**Comment by ArneBachmann** [comment]
I see, all tests are in the travis script.
Should be extracted into own file and invoked from there.

File: git_of_theseus/analyze.py [hunk 1]
```diff
@@ -21,37 +21,46 @@
...
 
 
-def analyze():
+def analyze(cohortfm=None, interval=None, ignore=None, only=None, outdir=None, branch=None, all_filetypes=None, repos=None):
     default_filetypes = set()
     for _, _, filetypes, _ in pygments.lexers.get_all_lexers():
         default_filetypes.update(filetypes)
     default_filetypes.difference_update(IGNORE_PYGMENTS_FILETYPES)
 
     parser = argparse.ArgumentParser(description='Analyze git repo')
```
Comment (erikbern) [inline git_of_theseus/analyze.py]: can you move the argument parser out of this function? can put it under `if __name__ == '__main__'` or in a separate function.

File: git_of_theseus/stack_plot.py [hunk 2]
```diff
@@ -32,45 +32,52 @@ def euclidean(a, b):
     return colors
 
 
-def stack_plot():
+def stack_plot(display=None, outfile=None, max_n=None, normalize=None, dont_stack=None, inputs=None):
     parser = argparse.ArgumentParser(description='Plot stack plot')
```
Comment (erikbern) [inline git_of_theseus/stack_plot.py]: same here – great if you can move the argument parser out

File: git_of_theseus/survival_plot.py [hunk 3]
```diff
@@ -21,20 +21,26 @@
 
 from matplotlib import pyplot
 
-def survival_plot():
+def survival_plot(exp_fit=None, display=None, outfile=None, years=None, inputs=None):
     parser = argparse.ArgumentParser(description='Plot survival plot')
```
Comment (erikbern) [inline git_of_theseus/survival_plot.py]: same

**Comment by erikbern** [comment]
looks great but would prefer if you can factor out the argument parsers out of the functions (since command line integration doesn't quite belong inside the functions if they are meant to be used as a library)

**Comment by ArneBachmann** [comment]
Totally valid point, but I did the first step and expect not to do all work myself, this is a collaboration platform...
If you want your code cleaner, then please improve it before merging :+1:

**Comment by erikbern** [comment]
thanks @ArneBachmann will make some improvements right now

**Pull request merged**